### PR TITLE
Add min stopping simulation option

### PR DESCRIPTION
**Pull request by govertv**
The suggested changes allow using 'min' as an option for --n_queries. When using 'min' the simulation run will stop when all relevant papers have been includes. My testing indicates it works correctly. However, the error message for an incorrect --n_queries value (not an int and not 'min') is now a default python error which might be not be readable for users.

Another change will also be needed to enable ASReview-visualization to draw the full recall curve (see: https://github.com/asreview/asreview/issues/598)

### CODE REVIEW

File: asreview/entry_points/simulate.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def _simulate_parser(prog="simulate", description=DESCRIPTION_SIMULATE):
              f"Default {DEFAULT_N_INSTANCES}.")
     parser.add_argument(
         "--n_queries",
-        type=int,
+        type=str,
```
Comment (J535D165) [inline asreview/entry_points/simulate.py:174]: I think it might be better to define a custom type over here that accepts integers and "min". Like described here: https://stackoverflow.com/questions/53445769/argparse-define-custom-actions-or-types-with-additional-arguments.

File: asreview/entry_points/simulate.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def _simulate_parser(prog="simulate", description=DESCRIPTION_SIMULATE):
...
         default=None,
         help="The number of queries. By default, the program "
```
Comment (J535D165) [inline asreview/entry_points/simulate.py:176]: Please add this option to the help. Don't forget to update this manually https://asreview.readthedocs.io/en/latest/API/cli.html#cmdoption-asreview-simulate-n-queries.

File: asreview/review/base.py [hunk 3]
```diff
@@ -258,9 +258,16 @@ def _stop_iter(self, query_i, n_pool):
...
-            stop_iter = True
+        # If n_queries is set to min, stop when all relevant papers are included
+        if self.n_queries == 'min':
+            n_included = np.count_nonzero(self.y[self.train_idx] == 1)
+            n_total_relevant = np.count_nonzero(self.y == 1)
+            if n_included == n_total_relevant:
+                stop_iter = True
+        # Otherwise, stop when reaching n_queries (if provided)
+        elif self.n_queries is not None:
+            if query_i >= int(self.n_queries): # Might be good to check for valid values somewhere earlier
```
Comment (J535D165) [inline asreview/review/base.py:269]: Agree. We can do this in another PR.

File: asreview/settings.py [hunk 4]
```diff
@@ -31,7 +31,7 @@
     "feature_extraction": str,
     "n_papers": int,
     "n_instances": int,
-    "n_queries": int,
+    "n_queries": str,
```
Comment (J535D165) [inline asreview/settings.py:34]: Can we use the same action as in the first comment?

**Review by J535D165 — CHANGES_REQUESTED** [review]
Nice PR, looking forward to merging this! Please have a look at the remarks and please fix the lint errors.

**Comment by J535D165** [comment]
I'm planning to merge soon. Do you have a plot of the recall curve in the visualization package when using this feature (as is)?

**Comment by J535D165** [comment]
As well as a comparison between `asreview stats` before and after this PR.

**Comment by govertv** [comment]
Here is a plot of the recall curve when using the min stopping feature, as generated by the visualization plugin:

min_stop_recall

After the PR, the stat plugin can correctly calculate statistics for state files generated with the min stopping option enabled. However, using these same state files in asreview stat before the PR results in an error ('min' is not a valid integer).

File: asreview/entry_points/simulate.py [hunk 5]
```diff
@@ -171,11 +172,12 @@ def _simulate_parser(prog="simulate", description=DESCRIPTION_SIMULATE):
...
     parser.add_argument(
         "--n_queries",
-        type=int,
+        type=type_n_queries,
         default=None,
-        help="The number of queries. By default, the program "
-             "stops after all documents are reviewed or is "
-             "interrupted by the user."
+        help="The number of queries. Alternatively, entering 'min' will stop the "
+             "simulation when all relevant documents have been found. By default, "
```
Comment (Rensvandeschoot) [inline asreview/entry_points/simulate.py:178]: Can you change ' documents'  into 'records'? See our vocabulary: https://asreview.readthedocs.io/en/latest/intro/vocabulary.html

File: docs/source/API/cli.rst [hunk 6]
```diff
@@ -181,8 +181,9 @@ Examples:
 
 .. option:: --n_queries N_QUERIES
 
-    The number of queries. By default, the program stops after all documents are reviewed
-    or is interrupted by the user.
+    The number of queries. Alternatively, entering :code:`min` will stop the simulation when all relevant
+    documents have been found. By default, the program stops after all documents are reviewed
```
Comment (Rensvandeschoot) [inline docs/source/API/cli.rst:185]: Can you change ' documents'  into 'records'? See our vocabulary: https://asreview.readthedocs.io/en/latest/intro/vocabulary.html

**Review by Rensvandeschoot — CHANGES_REQUESTED** [review]

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
A reproducible example that WSS and RRF are the same. 

Without `--n_queries`:
```
jonathan$ asreview simulate benchmark:Appenzeller-Herzog_2020 -s appherzog.h5 --init_seed 535 --seed 535

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

The following records are prior knowledge:

2187 - Serum transaminases in children with Wilson's disease                              Iorio, R.;D'Ambrosi, M.;Marcellini, M...
819 - Penicillamine-induced dermatomyositis. A case history                              Lund, H. I.;Nielsen, M.                 

```

```
jonathan$ asreview stat appherzog.h5 
************  appherzog.h5  **************

-----------  general  -----------
Number of runs            : 1
Number of papers          : 3453
Number of included papers : 29
Number of excluded papers : 3424
Number of unlabeled papers: 0
Number of queries         : 3452

-----------  settings  -----------
data_name         : Appenzeller-Herzog_2020
model             : nb
query_strategy    : max
balance_strategy  : double
feature_extraction: tfidf
n_instances       : 1
n_prior_included  : 1
n_prior_excluded  : 1
mode              : simulate
model_param       : {'alpha': 3.822}
query_param       : {}
feature_param     : {'ngram_max': 1, 'stop_words': 'english', 'split_ta': 0, 'use_keywords': 0}
balance_param     : {'a': 2.155, 'alpha': 0.94, 'b': 0.789, 'beta': 1.0}

-----------    ATD    -----------
 0.0762

Time to discovery:

...

-----------  WSS/RRF  -----------
WSS@95 : 50.59 %
WSS@100: 50.83 %
RRF@5  : 71.43 %
RRF@10 : 85.71 %
```

With `--n_queries min`:

```
jonathan$ asreview simulate benchmark:Appenzeller-Herzog_2020 -s appherzog_min.h5 --n_queries min --init_seed 535 --seed 535

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

The following records are prior knowledge:

2187 - Serum transaminases in children with Wilson's disease                              Iorio, R.;D'Ambrosi, M.;Marcellini, M...
819 - Penicillamine-induced dermatomyositis. A case history                              Lund, H. I.;Nielsen, M.                 
```

```
jonathan$ asreview stat appherzog_min.h5 
************  appherzog_min.h5  **********

-----------  general  -----------
Number of runs            : 1
Number of papers          : 3453
Number of included papers : 29
Number of excluded papers : 3424
Number of unlabeled papers: 0
Number of queries         : 1698

-----------  settings  -----------
data_name         : Appenzeller-Herzog_2020
model             : nb
query_strategy    : max
balance_strategy  : double
feature_extraction: tfidf
n_instances       : 1
n_queries         : min
n_prior_included  : 1
n_prior_excluded  : 1
mode              : simulate
model_param       : {'alpha': 3.822}
query_param       : {}
feature_param     : {'ngram_max': 1, 'stop_words': 'english', 'split_ta': 0, 'use_keywords': 0}
balance_param     : {'a': 2.155, 'alpha': 0.94, 'b': 0.789, 'beta': 1.0}

-----------    ATD    -----------
 0.0762

Time to discovery:

...

-----------  WSS/RRF  -----------
WSS@95 : 50.59 %
WSS@100: 50.83 %
RRF@5  : 71.43 %
RRF@10 : 85.71 %

```

**Comment by J535D165** [comment]
Congratz @govertv on this work!

@GerbrichFerdinands your feature request is finally there ;)

**Comment by Rensvandeschoot** [comment]
@govertv This is a very nice feature!!!

**Comment by GerbrichFerdinands** [comment]
@govertv really cool! great work!
